pkg/util/iptables/iptables.go

```diff
@@ -575,3 +576,235 @@ func IsNotFoundError(err error) bool {
 	}
 	return false
 }
+
+type RuleOption struct {
+	Arg     string
+	Negated bool
+}
+
+type Rule struct {
+	Modules []string
+	Jump    Chain
+	Comment string
+	Options map[string]RuleOption
+}
+
+type knownOption struct {
+	option      string
+	argRequired bool
+	canNegate   bool
+}
+
+func parseRule(line string, filterChains []Chain, filterChainPrefixes []string) (*Rule, Chain, error) {
+	knownOptions := []knownOption{
```

Comment:
But that means it's going to be constructing the array every time you call the function. If you made it global it would only have to do it once.